### DISCUSSION TITLE
How to properly configure Postfix to send email from a BB Docker container

### DISCUSSION THREAD

**Discussion by jdcaprace** [Budibase/budibase#3766] [Help]
Dear all,

Case: Self-host of BB.

I would like to configure properly my postfix install on my host to be able to properly send Emails for all my containers including BB containers. Actually, when I try to configure the SMTP in BB  (see image below) I got the following error:
![image](https://user-images.githubusercontent.com/12702125/145688601-2690d340-0d5b-4048-8756-79082b4f68b7.png)
BBWORKER container returns me: ERROR [1639249405317] (50 on d1c0f18d0f8b): connect ECONNREFUSED 127.0.0.1:25
My postfix can send email through my SMTP Gmail account without any problem using: mail -s "Test Email" myemail@gmail.com

I tried to adjust the config of the /etc/postfix/main.cf:
- Adding the IP of docker0 in the **inet_interfaces** variable
- Adding the private IP of BBWORKER container in the **mynetworks** variable

Until now, no success.
I would like to receive feedback to guide me on this config.
Thank you in advance.
Regards.

**Comment by mjashanks** [comment]
Hi Jean-David.

The budibase server is running inside a docker container. Therefore `localhost` refers to the container, not the host.

The host is available on..
- `172.17.0.1 ` for Linux
- `host.docker.internal` for mac and Windows

Hopefully that should fix your issue.

**Reply by jdcaprace** [reply]
Thank you so much for your quick answer Michael!

I managed now to configure properly my postfix to send Email on behalf of the docker containers.
For those like me that may suffer the same difficulty, I will share the key points here.
In BB configure the SMTP Email like this:
![image](https://user-images.githubusercontent.com/12702125/145691852-eb65918c-db2f-47d2-8a5c-e29036c5d3a7.png)
Then in your /etc/postfix/main.cf file verify your following variables:
"inet_interfaces = 127.0.0.1, 172.17.0.1" should have the IPs corresponding of the docker0.
"mynetworks = 127.0.0.0/8, [::1]/128, 172.17.0.0/24"  should have subnet of the docker0.

Hope that help!

**Reply by mjashanks** [reply]
Thank you for sharing! 🙂

**Reply by starmomou** [reply]
Hi~ same case here. I have no idea how to add the correct SMTP. Do I need postfix service using SMTP? Thank you
<img width="875" alt="螢幕截圖 2021-12-13 下午12 00 31" src="https://user-images.githubusercontent.com/73826858/145751512-cd9119fa-a565-464a-b686-cd760f5a7d42.png">

**Reply by mjashanks** [reply]
@starmomou Postfix is a self-host email server. Most people will use a 3rd party email service provider, such as postmark or SendGrid. The cloud providers have their own as well. Each service will have its own settings which you will use to configure the settings in your screenshot.

For example  - I use postmark. Their instructions are here: https://postmarkapp.com/developer/user-guide/send-email-with-smtp
